### ISSUE TITLE
socket TS011F manufacturer _TZ3000_cphmq0q7 not reporting anything except linkquality

### ISSUE THREAD

**Issue by slashbv** [Koenkk/zigbee2mqtt#9057]
# Bug Report
## What happened
I have this plug TS011F manufacturer _TZ3000_cphmq0q7 and it is not reporting power, current, voltage,energy and also it doesn't report the state when changed from physical button. The only things that it reports is linkquality.
![image](https://user-images.githubusercontent.com/59421121/136425104-ecaf5231-0e7a-4e3f-81b1-3eebe6d3e5e8.png)
![image](https://user-images.githubusercontent.com/59421121/136424688-ab649f3d-4d39-4bec-be80-6fc5ba8c23ee.png)

## Debug Info
Zigbee2MQTT version: 1.21.2
Adapter hardware: CC2652RB
Adapter firmware version: 20210708

**Comment by ristomatti** [comment]
I wonder if these need polling of values like with TS0121_plug? It would be rather unfortunate as I already got 8 of them but didn't like the fact they need polling. I bought these hoping for instantaneous reporting. 😄

**Comment by ristomatti** [comment]
I went through the pull requests at zigbee-herdsman and noticed some recent ones related to "_TZ3000_dpo1ysak". After upgrading to the latest-dev Docker image, indeed the values started to update, although it seems via polling. Also the device picture is now wrong but I guess this is what you occasionally get when buying stuff from AliExpress. 😄

Maybe it's the same for your device type @slashbv?

**Comment by slashbv** [comment]
I bought another one _TZ3000_cphmq0q7 that works fine. I suspect they might have different firmware.

**Comment by ristomatti** [comment]
Strange indeed. I wonder if there's any other identification the devices provide besides the model and manufacturer value that's just not visible on the Web UI?

**Comment by slashbv** [comment]
@Koenkk any chance to get a fix before the stable release ?

**Comment by Koenkk** [comment]
@slashbv try removing the `options: [exposes.options.measurement_poll_interval()],` line from the external converter.

**Comment by slashbv** [comment]
I have removed the line but Z2M is not starting anymore and the log is empty. docker logs shows this error:

Using '/app/data' as data directory
(node:7) UnhandledPromiseRejectionWarning: SyntaxError: Invalid or unexpected token
(Use `node --trace-warnings ...` to show where the warning was created)
(node:7) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:7) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by Koenkk** [comment]
Can you provide some more logging, I cannot reproduce it. Make sure to also update to z2m 1.22.0 (just released)

**Comment by slashbv** [comment]
Found the problem, I commented the line with # instead of // . Now it's fine it shows external converter in description but the physical switch still doesn't work. If I press the refresh button the state is updated.

**Comment by amdorofeev** [comment]
I have the same issue than @slashbv. I can read monitoring params only from dev console. Appversion 67, z2m 1.22.0

**Comment by Koenkk** [comment]
@slashbv my last hope is that the non-standard command send by the TuYa gateway sets up reporting (no 118 in your snif: https://github.com/Koenkk/zigbee2mqtt/issues/9057#issuecomment-950356998)

Can you:
- Update to the latest dev branch in 1 hour from now. (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)
- Use the following converter code:
```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    fingerprint: [{modelID: 'TS011F', manufacturerName: '_TZ3000_cphmq0q7', applicationVersion: 69}],
    model: 'TS011F_plug_3',
    description: 'Smart plug EXTERNAL CONVERTER',
    vendor: 'TuYa',
    fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.ignore_basic_report, fz.tuya_switch_power_outage_memory],
    toZigbee: [tz.on_off, tz.tuya_switch_power_outage_memory],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await endpoint.command('genBasic', 'tuyaSetup', {}, {disableDefaultResponse: true});
        endpoint.saveClusterAttributeKeyValue('haElectricalMeasurement', {acCurrentDivisor: 1000, acCurrentMultiplier: 1});
        endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
        device.save();
    },
    options: [exposes.options.measurement_poll_interval()],
    exposes: [e.switch(), e.power(), e.current(), e.voltage().withAccess(ea.STATE),
        e.energy(), exposes.enum('power_outage_memory', ea.STATE_SET, ['on', 'off', 'restore'])
            .withDescription('Recover state after power outage')],
};

module.exports = definition;
```
- Repair the device and press the yellow configure button in the frontend
  - If this doesn't work, provide a sniff when pairing this device to z2m + reconfiguring it via the frontend (make sure to also provide your network key here so that I can decrypt the sniff)

**Comment by slashbv** [comment]
unfortunately no change, same behavior.  I will sniff the traffic tomorrow.

**Comment by slashbv** [comment]
here is the sniff. I didn't change network key, it's the default one.
[wireshark_TZ3000_cphmq0q7_Z2M.zip](https://github.com/Koenkk/zigbee2mqtt/files/7464268/wireshark_TZ3000_cphmq0q7_Z2M.zip)

**Comment by Koenkk** [comment]
I don't see the expected command in the sniff, can you check wether configure is succesfull after clicking this? Start sniffing right before you click the configure button and provide the sniff here.

**Comment by slashbv** [comment]
what I did was to put the socket in pairing mode, start the sniff, enabled join in Z2M, once the socket was available in the frontend I have pressed the yellow button. The pairing was taking quite lot of time for some reason. I will do it again this evening and send you the sniff.

**Comment by slashbv** [comment]
I attached a new sniff only for pressing the yellow reconfigure button. I hope it's good this time.
[wireshark_TZ3000_cphmq0q7_Z2M_reconfigure.zip](https://github.com/Koenkk/zigbee2mqtt/files/7470975/wireshark_TZ3000_cphmq0q7_Z2M_reconfigure.zip)

**Comment by Koenkk** [comment]
Woops I made a mistake, turned out the change did not end up in the latest -dev, can you check again with the latest dev in 2 hours from now? (I would expect that you go an error when pressing the configure button??)

With the latest-dev pair the device + the reconfigure button. If that doesn't help make a sniff:
- start sniffing
- press configure button
- wait 10 seconds
- stop sniff.

**Comment by slashbv** [comment]
It doesn't work. Here is the sniff:
[wireshark_TZ3000_cphmq0q7_Z2M_reconfigure.zip](https://github.com/Koenkk/zigbee2mqtt/files/7479077/wireshark_TZ3000_cphmq0q7_Z2M_reconfigure.zip)

**Comment by slashbv** [comment]
there is also this error in the log now : 
Publish 'set' 'state' to 'socket_sara' failed: 'Error: Command 0x84fd27fffec2296f/1 genOnOff.off({}, {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":false,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Timeout - 9002 - 1 - 78 - 6 - 11 after 10000ms)'

**Comment by Koenkk** [comment]
@slashbv can you share a link where you bought this device? Then I can buy it also and debug furthe

**Comment by kbfifi** [comment]
I ordered 2 for 14 euro here: https://www.aliexpress.com/item/1005002605377621.html?spm=a2g0s.9042311.0.0.6dc34c4d2nqU3f
Pretty good deal I think.

Working reasonably since version 1.22.0 with CC2531. I noticed one issue possibly related to this issue: Data is not accurate when getting with topic:zigbee2mqtt/friendlyname/get payload: {"state":""} every 5 secs. It simply returns the old data until, I guess internally, every 60 seconds the data IS updated.

Also noted that when commanding topic:zigbee2mqtt/friendlyname/set payload: {"state":"ON"} the "state" property becomes visible. before I got: power, energy, voltage, link quality

**Comment by slashbv** [comment]
I can share the link no problem, but the thing is you might not receive it with the same software version as I have 2 and one works fine. Here is the link: https://www.aliexpress.com/item/1005002344798281.html?spm=a2g0s.9042311.0.0.48874c4d12ytjB

**Comment by slashbv** [comment]
I have just connected the working one to Tuya hub and guess what, it shows a firmware update. I suspect if I do it will be in the same situation as the other one.

**Comment by Koenkk** [comment]
@slashbv does the plug work after upading it?

**Comment by slashbv** [comment]
I am not going to update the firmware to the working one

**Comment by Koenkk** [comment]
Ah I misread, does the non-working one also show an update?

**Comment by slashbv** [comment]
non working one doesn't show any update. They both work well in with Tuya hub, physical button and reporting.

**Comment by Koenkk** [comment]
I've ordered the plug from https://www.aliexpress.com/item/1005002344798281.html?spm=a2g0s.9042311.0.0.48874c4d12ytjB , I have a TuYa gateway so I can update the fw. Will report back once I know more.

**Comment by pbaan93** [comment]
Hi, so i have been flowing this issue for a while now and now got the time to help, do you still need help? I have _TZ3000_mraovvmm form Blitzwolf 10x, updated 3 of them with the tuya gateway and the updated ones are not reporting in Z2M. The ones with the old firmware are work including reporting the state after manual on/off toggle. The detected model is "TS011F_plug_1" do you think we can assume they are all the same white label devices with the same firmware (and only a different sticker)? If you still need help: Where can I find a tutorial to do the snifing? I can compare the differences in the protocol, old vs updated firmware.

**Comment by oroszgergely** [comment]
If I press the physical OnOff button on the plug, it does not automatically reported to Z2M, it stays as it was, however i can read out the actual value on the dev console:
![image](https://user-images.githubusercontent.com/23665181/140620431-f4b65cde-81e4-417c-9899-7bdde2e3f3d5.png)
after the dev console readout, the state updates to the correct value immediately:
![image](https://user-images.githubusercontent.com/23665181/140620626-9fbce17b-028a-456b-9541-48bc60a4aced.png)



device info:
![image](https://user-images.githubusercontent.com/23665181/140620464-192575c8-8012-4e0b-bff5-0ed39d612593.png)

**Comment by Koenkk** [comment]
@pbaan93 I will check it once I receive this plug, if you want to give it a try, some links:
- https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#with-cc2531
- https://www.zigbee2mqtt.io/advanced/support-new-devices/02_support_new_tuya_devices.html#instructions
- https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html#requirements-and-caveats

**Comment by Koenkk** [comment]
Received the plug which came with `appVersion` `67`, paired it to the TuYa gateway and it showed an update (firmware `1.0.5`), after updating it got `appVersion` `69`. Using the TuYa gateway it indeed updates when you toggle it physically. I've sniffed that, paired the plug to z2m and replicated all the commands as far as I could, unfortunately no luck at all. TuYa did something quite nasty here. I therefore see only 2 possible options to fix this:
- TuYa releases an update for these plugs to fix this
- Someone with inside information about these TuYa devices helps us

**Comment by slashbv** [comment]
Thank you @Koenkk . How can we find someone with inside information ? :)  I hope Tuya will not do the same to more devices. Is there any way to go back to appVersion 67?

**Comment by kbfifi** [comment]
@Koenkk Do I understand correctly you can reproduce the problem of this issue with appVersion 69? Meaning appVersion is only producing linkquality info only? I own the exact same plug with version 69 and it's working fine for simple useage. I have another issue with it (#9536 )

**Comment by slashbv** [comment]
The reporting is not working either, if you read the history here, in order to get the values Koenkk chanced it to polling. You could probably change the polling interval. Your issue is the same as here.

**Comment by Koenkk** [comment]
This plug indeed doesn't support reporting so:
- Energy measurements are polled
- On/off state will not be updated when toggling physically

**Comment by rsettembrini** [comment]
I had just few second ago a chat with Tuya IoT Platform support. They said me that will handle the ticket. In short I asked a way to downgrade to app version 67 or a new upgrade version more compatible with zigbee2mqtt standards. Hoping in some good news in next days

**Comment by PM04290** [comment]
I have an unsupported state on my TS011F smartplug, model _TZ3000_cphmq0q7 is described in tuya.js of converter, you know why it stay not supported ?
i have a CC2531 key and Z2M version 1.18.1-1

**Comment by Koenkk** [comment]
> I have an unsupported state on my TS011F smartplug, model _TZ3000_cphmq0q7 is described in tuya.js of converter, you know why it stay not supported ? i have a CC2531 key and Z2M version 1.18.1-1

If you are still on 1.18.1 check first point of https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.19.0

**Comment by rsettembrini** [comment]
Hi Everyone,
I am in contact with a tuya engineer. He is asking me this: <<Hello, how did you integrate HA and your Zigbee2MQTT integration, Where did you see this? Could you please provide screenshots? Let me confirm (The previous app Version 67 worked fine in Homeassistant) what is this app Version 67

Would you better record a video of what you saw or obtained and send it to me>>

I said him that the device with the app version 67 worked fine, but it seems he doesn't understand what is the app version. 

Could you help me to give an answer him?
Thanks

**Comment by Koenkk** [comment]
@rsettembrini what we need to know is how to configure the device to make it send state updates when toggled physically.

**Comment by PM04290** [comment]
@Koenkk , my problem is solved. i was using the repository of danielwelch and not updated, when i have changed by the good repository, this one, everything worked.
I 'am not alone in this case, i publish a post on Facebook group and many people are on last repository.
i suggest to put a new release on the old repository to inform users to change.
best regards

**Comment by slashbv** [comment]
@PM04290 could you be more specific ?

**Comment by slashbv** [comment]
@rsettembrini the issue is with firmware 1.0.5. once you upgrade to this firmware reporting and state updates from physical button don't work anymore in Z2M.

**Comment by ake2018** [comment]
Hello, 
I wanted to ask if there are any updates regarding the behavior of the TS011F?
I can confirm that plugs with appVersion 69 do not send data, whilst plugs with AppVersion 67 do that.
I have both types of plugs and my temporary solution was the polling fix, but as mentioned by somebody else above, I get no update of the ON/OFF state of the device.

**Comment by Schmetterfliege** [comment]
@rsettembrini 
by chance any movement in the communication with them?
We will need a proper solution from the manufacturer, to stop this polling necessity :(

**Comment by rsettembrini** [comment]
@Schmetterfliege unfortunately no any update (see screenshot attached). they always answer they need a video or screenshot about firmware 1.0.5, but I really don't understand what they want. It seems taking time on their side at the moment .
![Cattura](https://user-images.githubusercontent.com/72879631/146587760-0c891cae-cefe-4fa7-9da0-ee0bcdff3a83.PNG)

**Comment by Schmetterfliege** [comment]
@rsettembrini 
Thank you very much for sharing!
To bad they don't seem to understand our query :(

I went ahead and create an account on this platform to log a ticket too...
Let's see how far I will get :D

**Comment by Schmetterfliege** [comment]
To give an update:
![image](https://user-images.githubusercontent.com/76066402/146772947-c9e17db0-6371-4f2e-ada7-276db368ca80.png)

As you can see... no update so far^^

**Comment by tammeryousef1006** [comment]
Dear all , i got the same plug and i had the same issues  , my method to fix it was by unpair it from zigbee2mqtt and pair it to tuya gateway hub , i did firmware upgrade then disconnect and wipe from the tuya app , after that pair it to zigbee2mqtt and it was reporting correctly , regarding the info pulling i set time to 5 seconds at settings tab at zigbee2mqtt front end , sorry if my answer wasnt clear

**Comment by Schmetterfliege** [comment]
@tammeryousef1006 
Maybe I'm wrong here, but that does **not** fix the issue, as you are still POLLING the data, rather than receiving it via PUSH...

**Comment by tammeryousef1006** [comment]
Sorry i had to delete the comment because i got different model TS0121 . As
you said it works only with pulling method

**Comment by Schmetterfliege** [comment]
I'm not sure if you could call this "progress", but we'll see where this goes:
![image](https://user-images.githubusercontent.com/76066402/146951404-77c7fe15-2244-499b-965b-4de0ba731cc9.png)

**Comment by Schmetterfliege** [comment]
Now that is a real and unfortunate update...
Let's see how far I get with imploring...
![image](https://user-images.githubusercontent.com/76066402/147129516-c226f328-2d0e-483c-9423-4deb4af99b22.png)

**Comment by Schmetterfliege** [comment]
@Koenkk 
This is chinese software; any chance you can read the private key or whatever on the gateway and fix this issue for us, in case Tuya doesn't cooperate? :D

**Comment by Koenkk** [comment]
@Schmetterfliege already tried: https://github.com/Koenkk/zigbee2mqtt/issues/9057#issuecomment-971992470

**Comment by Schmetterfliege** [comment]
> @Schmetterfliege already tried: [#9057 (comment)](https://github.com/Koenkk/zigbee2mqtt/issues/9057#issuecomment-971992470)

damn okay :(

**Comment by Schmetterfliege** [comment]
@Koenkk 
What exactly does the engineer mean with this sentence:
"Timed reporting DP is still standard zigbee 3.0 DP, so it can also be used on non-TUYA gateways"?
Wouldn't that mean that e.g. every 5 minutes we should still get updated values pushed from the device?

**Comment by Schmetterfliege** [comment]
Bad News:
![image](https://user-images.githubusercontent.com/76066402/147269300-a5737314-c3ca-4033-913d-519c0162d7f7.png)

**Comment by Koenkk** [comment]
Could you ask what a "private DP" means? If they just mean tuya manufacturer specific we should be able to handle it, the problem is that I don't see the device sending anything.

**Comment by Schmetterfliege** [comment]
Done - will let you know as soon as I get an answer :)

**Comment by Schmetterfliege** [comment]
@Koenkk 
This is the reply:

"Private DP" means non-zigbee 3.0 standard protocol, which is a private protocol added by tuya to the zigbee 3.0 standard protocol, and can be recognized by tuya gateways normally, but may not be recognized by non-tuya gateways

**Comment by Koenkk** [comment]
@Schmetterfliege since we can change whatever we want we can recognize this, the problem is that the device doesn't send anything (even not this private DP). I expect that some commands need to be send to this device in order to setup the reporting, could you ask if they know what?

**Comment by Schmetterfliege** [comment]
@Koenkk 
Can you send me the id that the dev requests?
Reply:
I'm sorry, but since your previous device is not connected to the tuya cloud, I can't get the information from my side, you can also ask the manufacturer of the product for help, or you can connect to the tuya cloud first and give me the virtual id on the tuya cloud so I can confirm the firmware logic.

**Comment by slashbv** [comment]
@Schmetterfliege you can use this one: bf8393ef7da5e2fa73jlwp

**Comment by Schmetterfliege** [comment]
@slashbv 
Thanks!

**Comment by Schmetterfliege** [comment]
Just to give a current status (since I'm not a patient person myself :) ):
The tuya contact wrote on dec. 25th:
"Roger, I will check the information, please be patient"

**Comment by Schmetterfliege** [comment]
Current status:
I requested an update and they replied that it might take some time for an update due to the upcoming chinese new year.

well.... anybody able to hack into their stuff and look up the info ourselves? xD

**Comment by nkrepo** [comment]
Is there any way to downgrade to a working firmware?

**Comment by Schmetterfliege** [comment]
@nkrepo not that I know of.
Blitzwolf would have to provide a new firmware/downgrade.
But I am in contact with Tuya, not Blitzwolf.

I have sent the "engineer" another message, requesting the information still before chinese new year...
I mean...it begins February 1st. That is still 27 days to go.
The event itself takes about 16 days.
-> no solution for at least 2 months from now on.
It's not like they'd have to change any code or develop something new... how hard can it be to look up the information for them to take more than 27 days for this...

E.g. I am looking at smart plugs worth ~ 200$, which I can only return 2 more weeks. 
Of course my message to them was more "friendly". Let's see how they react...

**Comment by Schmetterfliege** [comment]
@Koenkk 
The guy messaged me directly; not sure if this is something new (I doubt):
"Device data changes are not actively reported, the public version of the data should rely on the gateway to read the data policy"
Just in case, I copy pasted your message where you're suspecting that it requires a command to start the reporting.

**Comment by Schmetterfliege** [comment]
![image](https://user-images.githubusercontent.com/76066402/148544326-881b620b-59bd-414b-ae00-c242346a3fa3.png)

**Comment by nkrepo** [comment]
> ![image](https://user-images.githubusercontent.com/76066402/148544326-881b620b-59bd-414b-ae00-c242346a3fa3.png)
> 
> I told him that we don't see _anything_ coming from the device

Keep trying ...

**Comment by bobvandevijver** [comment]
Not sure if related, but the another tuya device I wrote an integration for uses a tuya private cluster, which can be configured for reporting by reading some properties from the basic cluster. That way it believes it is connected to the tuya cluster. You will need to know the configured datapoint though...

**Comment by Koenkk** [comment]
@bobvandevijver do you know what properties to read?

**Comment by bobvandevijver** [comment]
Actually, yes. I got this from the tuya guys:

- manufacturerName
- zclVersion
- appVersion
- modelId
- powerSource
- attributeReportingStatus

Not sure whether the order is important, but I request them in this order in one call on the Homey platform. 

After the call, it should push the values of all datapoints each minute on the tuya cluster (61184), or sooner if the value has just changed. 

Note that they actually use a command to report the data, it should be the data response one: https://developer.tuya.com/en/docs/iot-device-dev/tuya-zigbee-universal-docking-access-standard?id=K9ik6zvofpzql#subtitle-7-Private%20command%20id. The different data point contents are also on that page, but it looks like this:

```
dataResponse: {
    id: 0x1,
    args: {
      sequenceNumber: ZCLDataTypes.uint16,
      dataPoint: ZCLDataTypes.uint8,
      dataType: ZCLDataTypes.uint8,
      length: ZCLDataTypes.data16,
      data: ZCLDataTypes.buffer,
    },
  },
```

**Comment by Koenkk** [comment]
@bobvandevijver I'm familiar with all attributes except: `attributeReportingStatus`, do you know the attrid and type of it? It is not listed in the ZCL spec:

<img width="642" alt="Screenshot 2022-01-07 at 22 29 31" src="https://user-images.githubusercontent.com/2892853/148609886-92dfe2f6-4ac4-48d3-b2bb-5a561df647f4.png">

**Comment by Koenkk** [comment]
@bobvandevijver you are the man, I did some more sniffing with the Tuya gateway, this is the magic packet:

<img width="945" alt="Screenshot 2022-01-07 at 22 45 24" src="https://user-images.githubusercontent.com/2892853/148611461-c431c137-40d6-4994-b076-f5560ebed805.png">

After this physical control of the switch immediately updates the state, I will check about the power measurements

**Comment by Koenkk** [comment]
@bobvandevijver so checked further, after reading these attributes the plug starts to report on/off and energy, however voltage, current and power is not reported. Even checked this with the tuya gateway and I see the same behaviour here. The tuya gateway starts polling voltage, current and power once you open the "electric" screen, I don't see the plug reporting these values (even not via a tuya specific cluster). Do you see the same behaviour?

**Comment by git-developer** [comment]
A few days ago I received a delivery of BliTZWOlF BW-SHP15 which is affected of this issue. The manufacturer string is `_TZ3000_u5u4cakc`. I can confim that Koens change is working: using the `latest-dev` Docker image, the plug reports its state immediately. Current, energy, linkquality, power & voltage are logged once a minute. I changed `measurement_poll_interval` to 10 and 20, but these changes had no effect.

**Comment by wlcrs** [comment]
I also tested the changes and can confirm that the plugs are working as described bij @Koenkk . Many thanks @bobvandevijver for chiming in and giving the necessary clues to get this sorted!

**Comment by nkrepo** [comment]
@git-developer 

I tested the latest-dev, it reports energy, linkquality, power & voltage are logged once a minute but state is not reported correctly. I change the state from the physical button and the state remains unchanged and is not reported immediately. 
Am I missing something ?

**Comment by git-developer** [comment]
I don't think so - I just pulled and started the `latest-dev` docker image. Since that, every physical button press causes an MQTT message, for both a device that has been paired before and another one that wasn't. Maybe your device behaves different since it has a different manufacturer string.

**Comment by nkrepo** [comment]
My device is the same with @slashbv initial issue report.
The state is updated only when I hit refresh from the frontend, but not MQTT is message created with the physical button press.

![image](https://user-images.githubusercontent.com/8989693/148642729-e690ae0b-bd2c-470f-8846-415f16572ea1.png)

**Comment by bobvandevijver** [comment]
@Koenkk Good to hear that I could help! Found this issue by coincidence on [Tweakers](https://tweakers.net/nieuws/191710/tuya-budgetsmarthomeplatform-gaat-matter-ondersteunen.html?showReaction=17014132#r_17014132), so it is a bit lucky 🙂 I can only imagine you feel about the same I did when I finally got this answer (in Mandarin on WeChat) from the Chinese engineers I got in contact with through the whitelabeler I'm creating a Homey integration for!

By the way, `attributeReportingStatus` is one of those global attributes that should be available on each cluster (although this one is actually optional). 
 
![image](https://user-images.githubusercontent.com/1835343/148644858-63facde6-d362-4d6f-bce4-cf16d6d86fd9.png)

I do not have this particular device (I do have a `_TZ3000_5ity3zyu` / `TS0121`, although that once seemed to work fine with the normal Zigbee cluster).

The other device I have is a thermostat (`_TZE200_4hbx5cvx` / ``TS0601``): even though it exposes the ZCL thermostat cluster, it does not use it at all! All values are stuck on their defaults and you can write what you want without anything happening. The only way to set data from it is using their private cluster (`61184`), and the only way to retrieve data from it is after you've send this magic packet once so the values are being "reported"!

If the Tuya gateway for this device polls those properties you mention, I guess you're out of luck to get them to report. Although it might be that there is a data point value being reported with the value you're looking for, somewhere in the data being reported every minute on their private cluster. Unfortunately I cannot help you with how the datapoints are configured, as that officially is something only tuya and the whitelabeler should know (and I also do not have this device). You might be able to retrieve the configuration from the Tuya interface though, which some [developer tools digging}(https://www.zigbee2mqtt.io/advanced/support-new-devices/03_find_tuya_data_points.html) (but you should probably already know that, as I now see these docs are actually from you 😅)

**Comment by Koenkk** [comment]
@bobvandevijver thanks again !

Since we now have the same behaviour as when paired to the TuYa hub I think this can be closed now.

**Comment by nkrepo** [comment]
> @bobvandevijver thanks again !
> 
> Since we now have the same behaviour as when paired to the TuYa hub I think this can be closed now.

@Koenkk I think i am expiriencing a strange behaviour with TS011F /_TZ3000_cphmq0q7 reports energy, linkquality, power & voltage are logged once a minute but state is not reported correctly. I change the state from the physical button and the state remains unchanged and is not updated in the report, neither MQTT message is generated.

For example when I when I turn off from the physical button I continue to get in the report that the state is on.

`Zigbee2MQTT:info  2022-01-08 20:14:46: MQTT publish: topic 'zigbee2mqtt/smartplug_5', payload '{"current":0,"energy":0.01,"linkquality":69,"power":0,"power_outage_memory":null,"state":"ON","voltage":231}'
Zigbee2MQTT:info  2022-01-08 20:14:46: MQTT publish: topic 'zigbee2mqtt/smartplug_5', payload '{"current":0,"energy":0.01,"linkquality":65,"power":0,"power_outage_memory":null,"state":"ON","voltage":231}'`

**Comment by Koenkk** [comment]
@nkrepo did you update to the latest dev?  (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by andrei-trybukhouski** [comment]
>  When I change the state from the physical button and the state remains unchanged and is not updated in the report, neither MQTT message is generated.
> For example when I when I turn off from the physical button I continue to get in the report that the state is on.
I confirm the same behavior.
And when I switch with hard button, there is nothing in the logs.
docker, latest-dev, checked - commit with appropriate changes in place.

**Comment by nkrepo** [comment]
> 
@Koenkk 
I updated to today's latest-dev and I confirm that the same behavior continues.

**Comment by Koenkk** [comment]
@andrei-trybukhouski @nkrepo 

- Could you try to repair the device?
- Can you provide the data/database.db entry of this device?
- Do you have the original tuya gateway? Can you check if it works there.

**Comment by nkrepo** [comment]
@Koenkk 
I repaired the device and it started reporting:

` MQTT publish: topic 'zigbee2mqtt/smartplug_2', payload '{"current":0,"energy":0.11,"linkquality":91,"power":0,"power_outage_memory":null,"state":null,"voltage":233}'
`

My database entry for the device is:

`{"id":10,"type":"Router","ieeeAddr":"0x540f57fffe74b729","nwkAddr":50420,"manufId":4098,"manufName":"_TZ3000_cphmq0q7","powerSource":"Mains (single phase)","modelId":"TS011F","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":266,"inClusterList":[0,3,4,5,6,1794,2820,57344,57345],"outClusterList":[25,10],"clusters":{"genBasic":{"attributes":{"modelId":"TS011F","manufacturerName":"_TZ3000_cphmq0q7","powerSource":1,"zclVersion":3,"appVersion":69,"stackVersion":0,"hwVersion":1,"dateCode":""}},"haElectricalMeasurement":{"attributes":{"acCurrentDivisor":1000,"acCurrentMultiplier":1,"rmsVoltage":235,"rmsCurrent":0,"activePower":0}},"seMetering":{"attributes":{"divisor":100,"multiplier":1,"currentSummDelivered":[0,2]}},"genOnOff":{"attributes":{"onOff":1}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0021b77801","endpointID":1}],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":69,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{"configured":-1453151488},"lastSeen":1641731439766,"useImplicitCheckin":true}`

**Comment by Schmetterfliege** [comment]
Yesterday we've updated the Zigbee Adapter (for IoBroker) for these changes, I paired one of the plugs and it works as expected.
When switching the state via the physical button, I immediately see the new state.
power/current/voltage come in like every minute or so.

**Comment by andrei-trybukhouski** [comment]
unfortunately repair did not help
database entry:
`{"id":28,"type":"Router","ieeeAddr":"0x540f57fffe65d536","nwkAddr":56970,"manufId":4098,"manufName":"_TZ3000_cphmq0q7","powerSource":"Mains (single phase)","modelId":"TS011F","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":266,"inClusterList":[0,3,4,5,6,1794,2820,57344,57345],"outClusterList":[25,10],"clusters":{"genBasic":{"attributes":{"modelId":"TS011F","manufacturerName":"_TZ3000_cphmq0q7","powerSource":1,"zclVersion":3,"appVersion":69,"stackVersion":0,"hwVersion":1,"dateCode":""}},"haElectricalMeasurement":{"attributes":{"acCurrentDivisor":1000,"acCurrentMultiplier":1,"rmsVoltage":230,"rmsCurrent":0,"activePower":0}},"seMetering":{"attributes":{"divisor":100,"multiplier":1,"currentSummDelivered":[0,0]}},"genOnOff":{"attributes":{"onOff":1}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b001940c183","endpointID":1}],"configuredReportings":[{"cluster":6,"attrId":0,"minRepIntval":0,"maxRepIntval":300,"repChange":0}],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":69,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{"configured":-1453151488,"reporting":1},"lastSeen":1641736510411,"useImplicitCheckin":true}`

**Comment by nkrepo** [comment]
> Yesterday we've updated the Zigbee Adapter (for IoBroker) for these changes, I paired one of the plugs and it works as expected. When switching the state via the physical button, I immediately see the new state. power/current/voltage come in like every minute or so.
@Schmetterfliege 
Can you provide the db entry to check if there is any difference ?
I am using the Gio-dot Z-Bee Duo with CC2652P with [Koenkk 3.x.0 20211217](https://github.com/Koenkk/Z-Stack-firmware/blob/master/coordinator/Z-Stack_3.x.0/bin/CC1352P2_CC2652P_launchpad_coordinator_20211217.zip).

**Comment by Schmetterfliege** [comment]
@nkrepo 
can you show me how to get that db entry? Is it via a linux command?
With the zigbee adapter, all I'm doing is push the "pair mode" button, which will enable pairing for 60 seconds and then add the device :D
Not sure if this shows everything you/we would need :(
![image](https://user-images.githubusercontent.com/76066402/148694572-0b3cc940-8a5e-4bf3-8de4-8b3f8aa9a20d.png)

**Comment by Schmetterfliege** [comment]
oh an I'm using a conbee2 stick:
![image](https://user-images.githubusercontent.com/76066402/148694621-01f2bfd8-d503-4416-b23b-e5325003e22e.png)
Never updated it since I installed it ~ 1.5 years ago

**Comment by nkrepo** [comment]
@Schmetterfliege 
I see have your TS011F has different manufName:
My and  @andrei-trybukhouski 's have _TZ3000_cphmq0q7 and your has _TZ3000_u5u4cakc.
Maybe it has not the same firmware.

**Comment by Schmetterfliege** [comment]
@nkrepo 
Yea, saw that :/
But they are both listet in the same const. So the handling on herdsman side is the same for both.

Can you show me your path for this file? Maybe it makes it easier to find it on my pi :D

**Comment by nkrepo** [comment]
@Schmetterfliege 

If you followed the [linux guide](https://www.zigbee2mqtt.io/guide/installation/01_linux.html#determine-location-of-the-adapter-and-checking-user-permissions) for installation. The config files should be in:
`/opt/zigbee2mqtt/data/`

**Comment by Schmetterfliege** [comment]
> 

Well, I installed iobroker. It installs all those things by itself.
It has installed herdsman:
![image](https://user-images.githubusercontent.com/76066402/148697274-15de7164-26c2-4155-8c40-aa80be00350b.png)

But I can't find any zigbee2mqtt folder anywhere :/

**Comment by bobvandevijver** [comment]
> _TZ3000_cphmq0q7 and yours has _TZ3000_u5u4cakc

Note that tuya uses the manufacturer id as device identifier as well as the actual device id, so they can indeed be quite different.

**Comment by nkrepo** [comment]
In https://github.com/Koenkk/zigbee2mqtt/issues/9493#issuecomment-1004222318 the same issue is described. 
The state value is updated only after querying it.
In our case also no MQTT message is generated when physical button is pressed.

**Comment by nkrepo** [comment]
@Koenkk is there anything I can to do to assist with the debugging ?

**Comment by Koenkk** [comment]
@nkrepo could you provide me the herdsman debug log when pressing configure (yellow refresh icon) in the frontend?

Once this is done, wait for 5 seconds and toggle the switch physically

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

**Comment by nkrepo** [comment]
@Koenkk 
Here are the logs, the device under test is the smartplug_2.
I am using zigbee2mqtt through docker so I got the stdout for the device via 
`docker logs zigbee2mqtt |grep smartplug_2`

I hit the reconfigure in the frontend and after after a phew seconds the physical button, no message was generated when i hit the physical button and the state was not updated.
The state is only updated when I hit the state refresh button in the frontend, after the:
`Publishing get 'get' 'state' to 'smartplug_2'`


[zigbee2mqtt.log](https://github.com/Koenkk/zigbee2mqtt/files/7841314/zigbee2mqtt.log)

**Comment by Koenkk** [comment]
@nkrepo this only contains the normal debug logging, not the herdsman debug logging, this logging is only logged to STDOUT. https://www.zigbee2mqtt.io/guide/usage/debug.html#zigbee-herdsman-debug-logging

**Comment by nkrepo** [comment]
> @nkrepo this only contains the normal debug logging, not the herdsman debug logging, this logging is only logged to STDOUT. https://www.zigbee2mqtt.io/guide/usage/debug.html#zigbee-herdsman-debug-logging

@Koenkk 
Sorry my mistake i forgot to recreate the containers.
Here are the full logs

[zigbee2mqtt.log](https://github.com/Koenkk/zigbee2mqtt/files/7841645/zigbee2mqtt.log)

**Comment by elzershark** [comment]
[wireshark (1).zip](https://github.com/Koenkk/zigbee2mqtt/files/7842257/wireshark.1.zip)
Ich weiß nicht, ob man das gebrauchen kann. Das habe ich  mit wireshark aufgezeichnet.
Das wird gesendet von der Steckdose zu der tuya zigbee bridge, wenn ich die Steckdose am Taster selber schalte.

**Comment by nkrepo** [comment]
@Koenkk are the logs OK , is the physical button press recorded?

@elzershark How you grab the traffic between zigbee device and tuya bridge with wireshark?

**Comment by elzershark** [comment]
yes, physical button. i have make this.. https://www.zigbee2mqtt.io/advanced/zigbee/04_sniff_zigbee_traffic.html#prerequisites for windows with Wireshark. then i have paired the socked with tuya bridge and i have testet the ports vor tuya bridget. port +1 and press button. then, i think port 6, i have the log

**Comment by Koenkk** [comment]
@nkrepo seems the workaround doesn't work for your plug.

- Could you provide the data/database.db entry of your device?
- Could you provide a sniff when pairing the plug to the original tuya gateway and toggling it physically afterwards?

**Comment by nkrepo** [comment]
@Koenkk 

The database entry of my device is 
`{"id":9,"type":"Router","ieeeAddr":"0x540f57fffe684224","nwkAddr":64970,"manufId":4098,"manufName":"_TZ3000_cphmq0q7","powerSource":"Mains (single phase)","modelId":"TS011F","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":266,"inClusterList":[0,3,4,5,6,1794,2820,57344,57345],"outClusterList":[25,10],"clusters":{"genBasic":{"attributes":{"65503":"\u0000\u0000\u0000\u0000\u0005\u0000\u0000\u0000\u0000\u0005\u0000\u0000\u0000\u0000\u0012\u0000\u0000\u0000\u0000\u0005\u0000\u0000\u0000\u0000\u0005!�m)\u0012","65506":31,"65508":0,"65534":0,"modelId":"TS011F","manufacturerName":"_TZ3000_cphmq0q7","stackVersion":0,"dateCode":"","zclVersion":3,"appVersion":69,"powerSource":1}},"haElectricalMeasurement":{"attributes":{"acCurrentDivisor":1000,"acCurrentMultiplier":1,"rmsVoltage":229,"rmsCurrent":0,"activePower":0}},"seMetering":{"attributes":{"divisor":100,"multiplier":1,"currentSummDelivered":[0,12]}},"genOnOff":{"attributes":{"onOff":0}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00124b0021b77801","endpointID":1}],"configuredReportings":[],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":69,"stackVersion":0,"hwVersion":1,"dateCode":"","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1121114317},"lastSeen":1641925349376,"useImplicitCheckin":true}`

I don't have the original tuya gateway and I cannot sniff it, @elzershark probably can help with it.

**Comment by Schmetterfliege** [comment]
small remark with regards to @elzershark's comment:
My device also shows no log message when triggering it physically, but the state successfully changes.
When triggering it via the zigbee adapter in iobroker, I see this log entry:
`zigbee.0 2022-01-08 23:07:14.263 warn SetState_typed called without type`

**Comment by elzershark** [comment]
open this in wireshark.. this is the log...
Wireshark (1).zip

**Comment by Koenkk** [comment]
@elzershark this sniff is incomplete, I need a sniff starting from the point where you put the TuYa gateway into pairing mode, pair the plug and after that control it physically.

**Comment by elzershark** [comment]
@Koenkk  ok.. i have.
[paired_tuya.zip](https://github.com/Koenkk/zigbee2mqtt/files/7870300/paired_tuya.zip)
.

**Comment by Koenkk** [comment]
@elzershark @Schmetterfliege could you check if this issue has been fixed with the following external converter:

```js
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('../lib/tuya');

const definition = {
    fingerprint: [].concat(...TS011Fplugs.map((manufacturerName) => {
        return [69, 68, 65, 64].map((applicationVersion) => {
            return {modelID: 'TS011F', manufacturerName, applicationVersion};
        });
    })),
    model: 'TS011F_plug_3',
    description: 'Smart plug (with power monitoring by polling) CUSTOM',
    vendor: 'TuYa',
    whiteLabel: [{vendor: 'VIKEFON', model: 'TS011F'}, {vendor: 'BlitzWolf', model: 'BW-SHP15'}],
    fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.ignore_basic_report, fz.tuya_switch_power_outage_memory],
    toZigbee: [tz.on_off, tz.tuya_switch_power_outage_memory],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        // Enables reporting of physical state changes
        // https://github.com/Koenkk/zigbee2mqtt/issues/9057#issuecomment-1007742130
        await endpoint.read('genBasic', ['manufacturerName', 'zclVersion', 'appVersion', 'modelId', 'powerSource', 0xfffe]);
        await endpoint.write('genBasic', {0xffde: {value: 0x13, type: 0x20}});
        await endpoint.read('genOnOff', [0x0000, 0x4001, 0x4002, 0x5000, 0x8000, 0x8001, 0x8002, 0x8003]);
        endpoint.saveClusterAttributeKeyValue('haElectricalMeasurement', {acCurrentDivisor: 1000, acCurrentMultiplier: 1});
        endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
        device.save();
    },
    options: [exposes.options.measurement_poll_interval()],
    exposes: [e.switch(), e.power(), e.current(), e.voltage().withAccess(ea.STATE),
        e.energy(), exposes.enum('power_outage_memory', ea.STATE_SET, ['on', 'off', 'restore'])
            .withDescription('Recover state after power outage')],
    onEvent: tuya.onEventMeasurementPoll,
};

module.exports = definition;
```

- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m, go to the frontend, verify that the description of the device ends with CUSTOM (this means that external converter has been loaded correctly)
- press the yellow reconfigure button
- does it report the status now?

**Comment by Diondk** [comment]
I updated to the lastest Dev and my manual press on the shp15 is now being updated automatically

**Comment by Schmetterfliege** [comment]
@Koenkk 
Mine is working fine I think.
I'm using the zigbee adapter in iobroker, and didn't find the zigbee2mqtt module/folder not yet on my raspberry pi :/

**Comment by Koenkk** [comment]
@Schmetterfliege with this external converter it is working fine? Or just with the latest io broker?

**Comment by Schmetterfliege** [comment]
> 

The dev of the zigbee adapter had updated to the latest zigbee-herdsman after I requested it (after your changes! :) )
So with the latest herdsman it worked perfectly fine.

But as someone mentioned above: it seems to be related to the fingerprint whether it works fine with that or if it still has issues :S

**Comment by elzershark** [comment]
yes, thats right @Schmetterfliege .
i have upgrade. ( npm install zigbee-herdsman-converters -u iobroker)
and the button works.

when i put this: 
```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('../lib/tuya');

const definition = {
    fingerprint: [].concat(...TS011Fplugs.map((manufacturerName) => {
        return [69, 68, 65, 64].map((applicationVersion) => {
            return {modelID: 'TS011F', manufacturerName, applicationVersion};
        });
    })),
    model: 'TS011F_plug_3',
    description: 'Smart plug (with power monitoring by polling) CUSTOM',
    vendor: 'TuYa',
    whiteLabel: [{vendor: 'VIKEFON', model: 'TS011F'}, {vendor: 'BlitzWolf', model: 'BW-SHP15'}],
    fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering, fz.ignore_basic_report, fz.tuya_switch_power_outage_memory],
    toZigbee: [tz.on_off, tz.tuya_switch_power_outage_memory],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        // Enables reporting of physical state changes
        // https://github.com/Koenkk/zigbee2mqtt/issues/9057#issuecomment-1007742130
        await endpoint.read('genBasic', ['manufacturerName', 'zclVersion', 'appVersion', 'modelId', 'powerSource', 0xfffe]);
        await endpoint.write('genBasic', {0xffde: {value: 0x13, type: 0x20}});
        await endpoint.read('genOnOff', [0x0000, 0x4001, 0x4002, 0x5000, 0x8000, 0x8001, 0x8002, 0x8003]);
        endpoint.saveClusterAttributeKeyValue('haElectricalMeasurement', {acCurrentDivisor: 1000, acCurrentMultiplier: 1});
        endpoint.saveClusterAttributeKeyValue('seMetering', {divisor: 100, multiplier: 1});
        device.save();
    },
    options: [exposes.options.measurement_poll_interval()],
    exposes: [e.switch(), e.power(), e.current(), e.voltage().withAccess(ea.STATE),
        e.energy(), exposes.enum('power_outage_memory', ea.STATE_SET, ['on', 'off', 'restore'])
            .withDescription('Recover state after power outage')],
    onEvent: tuya.onEventMeasurementPoll,
};

module.exports = definition;
```

that is ok, but in iobroker log is this:

`DeviceConfigure:DeviceConfigure failed 0x540f57fffefdc1ef TS011F, attempt 4 (Error: Read 0x540f57fffefdc1ef/1 genOnOff([0,16385,16386,20480,32768,32769,32770,32771], {"sendWhenActive":false,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE') at Endpoint.checkStatus (/opt/iobroker/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:310:23) at Endpoint.read (/opt/iobroker/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:430:22) at Object.configure (/opt/iobroker/node_modules/zigbee-herdsman-converters/devices/tuya.js:1160:13) at DeviceConfigure.doConfigure (/opt/iobroker/node_modules/iobroker.zigbee/lib/zbDeviceConfigure.js:145:9) at DeviceConfigure.configure (/opt/iobroker/node_modules/iobroker.zigbee/lib/zbDeviceConfigure.js:124:17))`

So upgrading is enough. thank you

**Comment by Koenkk** [comment]
Great, then this can be closed 🥳

**Comment by Schmetterfliege** [comment]
@Koenkk 
@nkrepo still had issues with his device, not sure how the current state is on his/her side though :)

**Comment by nkrepo** [comment]
@Koenkk and @Schmetterfliege the physical button state is reported correctly with latest dev and a mqtt message is generated for every physical button press.

I have not tested yet if the #9987 applies also to _TZ3000_cphmq0q7.

**Comment by RutgerDiehard** [comment]
I've just updated 5 _TZ3000_cphmq0q7 plugs to 1.0.5 via the Tuya gateway and SmartLife app to try and fix random switch offs as experienced by others. The switch offs seem to have been fixed but none of the devices are showing Voltage, Current or Power. I understand that his has been fixed in v1.22 of zigbee2mqtt although I've not been able to get the plugs to work successfully even though I'm running 1.27.2-1 of the HA addon. I've force-removed, added and reconfigured but still no success. Is there something else I need to do?
